Lib/test/test_generators.py

```diff
@@ -270,6 +270,23 @@ def gen():
         self.assertEqual(next(g), "done")
         self.assertEqual(sys.exc_info(), (None, None, None))
 
+    def test_except_throw_bad_exception(self):
+        class E(Exception):
+            def __new__(cls, *args, **kwargs):
+                return cls
+
+        def boring_generator():
+            yield
+
+        gen = boring_generator()
+
+        with self.assertRaisesRegex(
+                TypeError,
+                'should have returned an instance of BaseException'):
+            gen.throw(E)
+
+        self.assertRaises(StopIteration, next, gen)
```

Comment:
Now that we know the generator is exhausted, the next question might be "does the generator catch the `TypeError`"?

So perhaps:
```python
def boring_generator():
    with self.assertRaisesRegex(
            TypeError,
            'should have returned an instance of BaseException'):
        yield

gen = boring_generator()
next(gen)  # advance to the yield in the with statement
gen.throw(E)
self.assertRaises(StopIteration, next, gen)
```